app/src/main/java/com/ustwo/boilerplate/base/BaseActivity.java

```diff
@@ -29,4 +30,6 @@ protected void onDestroy() {
 
   @NonNull
   protected abstract V getPresenterView();
+
+  protected abstract C createComponentAndInject();
```

Comment:
I think you can call createComponentAndInject() method from onCreate() and get rid of it from the children of the activity